crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -2364,6 +2426,25 @@ impl<
         );
     }
 
+    fn unbox_str_or_list(
+        buf: &mut Vec<'a, u8>,
+        storage_manager: &mut StorageManager<'a, 'r, GeneralReg, FloatReg, ASM, CC>,
+        dst: Symbol,
+        ptr_reg: GeneralReg,
+        tmp_reg: GeneralReg,
+    ) {
+        let base_offset = storage_manager.claim_stack_area(&dst, 24);
+
+        ASM::mov_reg64_mem64_offset32(buf, tmp_reg, ptr_reg, 0);
+        ASM::mov_base32_reg64(buf, base_offset, tmp_reg);
+
+        ASM::mov_reg64_mem64_offset32(buf, tmp_reg, ptr_reg, 8);
+        ASM::mov_base32_reg64(buf, base_offset + 8, tmp_reg);
+
+        ASM::mov_reg64_mem64_offset32(buf, tmp_reg, ptr_reg, 16);
+        ASM::mov_base32_reg64(buf, base_offset + 16, tmp_reg);
+    }
```

Comment:
maybe this can be generalized? anyway I found it helpful to write it out in full myself.

Reply:
This is basically copying from heap to stack. So it should essentially be the same as the method that copies from stack location to stack location, just with a real pointer instead of 2 stack offsets. So yeah, should be able to share code in general.